nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/diagnostics/DiagnosticUtils.java

```diff
@@ -19,9 +19,22 @@
 /**
  */
 public final class DiagnosticUtils {
+    final private static String OperatingSystem = System.getProperty("os.name").toLowerCase();
 
     public static int getUtilization(final double used, final double total) {
         return (int) Math.round((used / total) * 100);
     }
 
+    public static boolean isWindows() {
+        return OperatingSystem.contains("win");
+    }
+
+    public static boolean isMac() {
+        return OperatingSystem.contains("mac");
+    }
+
+    public static boolean isLinuxUnix() {
+        return OperatingSystem.contains("nix") ||OperatingSystem.contains("nux") || OperatingSystem.contains("aix");
+    }
+
```

Comment:
These methods can be replaced with use of Commons Lang3 [SystemUtils](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemUtils.html)